c7n/resources/rdsparamgroup.py

```diff
@@ -0,0 +1,339 @@
+            name = self.get_pg_name(param_group)
+            try:
+                self.do_delete(client, name)
+            except ClientError:
+                # TODO - anything we need to catch?
+                raise
+
+            self.log.info('Deleted RDS parameter group: %s', name)
+
+
+@pg_actions.register('delete')
+class PGDelete(PGMixin, Delete):
+    """Action to delete an RDS parameter group
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: rds-param-group-delete
+                resource: rds-param-group
+                filters:
+                  - DBParameterGroupName: pg_name
+                actions:
+                  - type: delete
+    """
+
+    permissions = ('rds:DeleteDBParameterGroup',)
+
+    def do_delete(self, client, name):
+        client.delete_db_parameter_group(DBParameterGroupName=name)
+
+
+@pg_cluster_actions.register('delete')
+class PGClusterDelete(PGClusterMixin, Delete):
+    """Action to delete an RDS cluster parameter group
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: rds-cluster-param-group-delete
+                resource: rds-cluster-param-group
+                filters:
+                  - DBClusterParameterGroupName: cluster_pg_name
+                actions:
+                  - type: delete
+    """
+
+    permissions = ('rds:DeleteDBClusterParameterGroup',)
+
+    def do_delete(self, client, name):
+        client.delete_db_cluster_parameter_group(DBClusterParameterGroupName=name)
+
+
+class Modify(BaseAction):
+
+    schema = type_schema(
+        'modify',
+        **{
+            'required': ['params'],
+            'params': {
+                'type': 'array',
+                'items': {
+                    'type': 'object',
+                    'required': ['name', 'value'],
+                    'name': {'type': 'string'},
+                    'value': {'type': 'string'},
+                    'apply-method': {'type': 'string', 'enum': ['immediate', 'pending-reboot']}
+                },
+            },
+        }
+    )
+
+    def process(self, param_groups):
+        client = local_session(self.manager.session_factory).client('rds')
+
+        params = []
+        for param in self.data.get('params', []):
+            params.append({
+                'ParameterName': param['name'],
+                'ParameterValue': param['value'],
+                'ApplyMethod': param.get('apply-method', 'immediate'),
+            })
+
+        # Can only do 20 elements at a time per docs, so if we have more than that we will
+        # break it into multiple requests: https://goo.gl/Z6oGNv
+        params = [iter(params)] * 20
+        params = itertools.izip_longest(*params, fillvalue={})
+
+        for param_set in params:
+            for param_group in param_groups:
+                name = self.get_pg_name(param_group)
+                try:
+                    self.do_modify(client, name, param_set)
```

Comment:
so we also need to check the value we're setting to is actually different then the value that's already there, if its not aka we have no modifications for the set of param changes, we shouldn't do the api call.